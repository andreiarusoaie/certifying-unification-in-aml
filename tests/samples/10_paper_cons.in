variables: x, z
symbols: cons, a
problem: cons(x, a) =? cons(a, z)
