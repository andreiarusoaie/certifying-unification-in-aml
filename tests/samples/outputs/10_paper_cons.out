Found Maude version: 3.0beta1

[1m[95mProof of:[94m (cons(a, z) or cons(x, a) <--> âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and cons(v(8), v(9)))  [0m
(1)(âˆƒ v(5) . (cons(x, a) <--> v(5) and (v(5) === cons(x, a)))) [e-subst] ;

(2)((cons(x, a) <--> âˆƒ v(5) . v(5) and (v(5) === cons(x, a))) <--> âˆƒ v(5) . (cons(x, a) <--> v(5) and (v(5) === cons(x, a)))) [e-scope] ;

(3)(cons(x, a) <--> âˆƒ v(5) . v(5) and (v(5) === cons(x, a))) [modus-ponens(1, 2)] ;

(4)(âˆƒ v(5) . (cons(a, z) <--> v(5) and (v(5) === cons(a, z)))) [e-subst] ;

(5)((cons(a, z) <--> âˆƒ v(5) . v(5) and (v(5) === cons(a, z))) <--> âˆƒ v(5) . (cons(a, z) <--> v(5) and (v(5) === cons(a, z)))) [e-scope] ;

(6)(cons(a, z) <--> âˆƒ v(5) . v(5) and (v(5) === cons(a, z))) [modus-ponens(4, 5)] ;

(7)(cons(a, z) or cons(x, a) <--> (âˆƒ v(5) . v(5) and (v(5) === cons(a, z))) or (âˆƒ v(5) . v(5) and (v(5) === cons(x, a)))) [eqv-or(3, 6)] ;

(8)((âˆƒ v(5) . v(5) and (v(5) === cons(a, z))) or (âˆƒ v(5) . v(5) and (v(5) === cons(x, a))) <--> âˆƒ v(5) . v(5) and (v(5) === cons(a, z)) or v(5) and (v(5) === cons(x, a))) [e-collapse] ;

(9)(v(5) and ((v(5) === cons(a, z)) or (v(5) === cons(x, a))) <--> v(5) and (v(5) === cons(a, z)) or v(5) and (v(5) === cons(x, a))) [and-or-distr] ;

(10)((âˆƒ v(5) . v(5) and ((v(5) === cons(a, z)) or (v(5) === cons(x, a)))) <--> âˆƒ v(5) . v(5) and (v(5) === cons(a, z)) or v(5) and (v(5) === cons(x, a))) [e-context(9)] ;

(11)(cons(a, z) or cons(x, a) <--> âˆƒ v(5) . v(5) and (v(5) === cons(a, z)) or v(5) and (v(5) === cons(x, a))) [eqv-tranz(7, 8)] ;

(12)(cons(a, z) or cons(x, a) <--> âˆƒ v(5) . v(5) and ((v(5) === cons(a, z)) or (v(5) === cons(x, a)))) [eqv-tranz(11, 10)] ;

(13)((v(5) === cons(x, a)) <--> âˆƒ v(8) ; v(9) . (v(5) === cons(v(8), v(9))) and (v(8) === x) and (v(9) === a)) [e-gen] ;

(14)((v(5) === cons(x, a)) <--> âˆƒ v(8) ; v(9) . (v(5) === cons(v(8), v(9))) and (v(8) === x) and (v(9) === a)) [and-eqv-intro(13)] ;

(15)((âˆƒ v(8) ; v(9) . (v(5) === cons(v(8), v(9))) and (v(8) === x) and (v(9) === a)) <--> âˆƒ v(8) ; v(9) . (v(5) === cons(v(8), v(9))) and (v(8) === x) and (v(9) === a)) [e-scope] ;

(16)((v(5) === cons(x, a)) <--> âˆƒ v(8) ; v(9) . (v(5) === cons(v(8), v(9))) and (v(8) === x) and (v(9) === a)) [eqv-tranz(14, 15)] ;

(17)((v(5) === cons(a, z)) <--> âˆƒ v(8) ; v(9) . (v(5) === cons(v(8), v(9))) and (v(8) === a) and (v(9) === z)) [e-gen] ;

(18)((v(5) === cons(a, z)) <--> âˆƒ v(8) ; v(9) . (v(5) === cons(v(8), v(9))) and (v(8) === a) and (v(9) === z)) [and-eqv-intro(17)] ;

(19)((âˆƒ v(8) ; v(9) . (v(5) === cons(v(8), v(9))) and (v(8) === a) and (v(9) === z)) <--> âˆƒ v(8) ; v(9) . (v(5) === cons(v(8), v(9))) and (v(8) === a) and (v(9) === z)) [e-scope] ;

(20)((v(5) === cons(a, z)) <--> âˆƒ v(8) ; v(9) . (v(5) === cons(v(8), v(9))) and (v(8) === a) and (v(9) === z)) [eqv-tranz(18, 19)] ;

(21)((v(5) === cons(a, z)) or (v(5) === cons(x, a)) <--> (âˆƒ v(8) ; v(9) . (v(5) === cons(v(8), v(9))) and (v(8) === a) and (v(9) === z)) or (âˆƒ v(8) ; v(9) . (v(5) === cons(v(8), v(9))) and (v(8) === x) and (v(9) === a))) [eqv-or(16, 20)] ;

(22)((âˆƒ v(8) ; v(9) . (v(5) === cons(v(8), v(9))) and (v(8) === a) and (v(9) === z)) or (âˆƒ v(8) ; v(9) . (v(5) === cons(v(8), v(9))) and (v(8) === x) and (v(9) === a)) <--> âˆƒ v(8) ; v(9) . (v(5) === cons(v(8), v(9))) and (v(8) === a) and (v(9) === z) or (v(5) === cons(v(8), v(9))) and (v(8) === x) and (v(9) === a)) [e-collapse] ;

(23)((v(5) === cons(a, z)) or (v(5) === cons(x, a)) <--> âˆƒ v(8) ; v(9) . (v(5) === cons(v(8), v(9))) and (v(8) === a) and (v(9) === z) or (v(5) === cons(v(8), v(9))) and (v(8) === x) and (v(9) === a)) [eqv-tranz(21, 22)] ;

(24)(((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (v(5) === cons(v(8), v(9))) <--> (v(5) === cons(v(8), v(9))) and (v(8) === a) and (v(9) === z) or (v(5) === cons(v(8), v(9))) and (v(8) === x) and (v(9) === a)) [and-or-distr] ;

(25)((âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (v(5) === cons(v(8), v(9)))) <--> âˆƒ v(8) ; v(9) . (v(5) === cons(v(8), v(9))) and (v(8) === a) and (v(9) === z) or (v(5) === cons(v(8), v(9))) and (v(8) === x) and (v(9) === a)) [e-context(24)] ;

(26)((v(5) === cons(a, z)) or (v(5) === cons(x, a)) <--> âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (v(5) === cons(v(8), v(9)))) [eqv-tranz(23, 25)] ;

(27)((âˆƒ v(5) . v(5) and ((v(5) === cons(a, z)) or (v(5) === cons(x, a)))) <--> âˆƒ v(5) . v(5) and (âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (v(5) === cons(v(8), v(9))))) [e-context(26)] ;

(28)(v(5) and (âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (v(5) === cons(v(8), v(9)))) <--> âˆƒ v(8) ; v(9) . v(5) and ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (v(5) === cons(v(8), v(9)))) [e-scope] ;

(29)((âˆƒ v(5) . v(5) and (âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (v(5) === cons(v(8), v(9))))) <--> âˆƒ v(5) . âˆƒ v(8) ; v(9) . v(5) and ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (v(5) === cons(v(8), v(9)))) [e-context(28)] ;

(30)((âˆƒ v(5) . âˆƒ v(8) ; v(9) . v(5) and ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (v(5) === cons(v(8), v(9)))) <--> âˆƒ v(5) ; v(8) ; v(9) . v(5) and ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (v(5) === cons(v(8), v(9)))) [e-set] ;

(31)((âˆƒ v(5) . v(5) and (âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (v(5) === cons(v(8), v(9))))) <--> âˆƒ v(5) ; v(8) ; v(9) . v(5) and ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (v(5) === cons(v(8), v(9)))) [eqv-tranz(29, 30)] ;

(32)(((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (âˆƒ v(5) . v(5) and (v(5) === cons(v(8), v(9)))) <--> âˆƒ v(5) . v(5) and ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (v(5) === cons(v(8), v(9)))) [e-scope] ;

(33)((âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (âˆƒ v(5) . v(5) and (v(5) === cons(v(8), v(9))))) <--> âˆƒ v(8) ; v(9) . âˆƒ v(5) . v(5) and ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (v(5) === cons(v(8), v(9)))) [e-context(32)] ;

(34)((âˆƒ v(8) ; v(9) . âˆƒ v(5) . v(5) and ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (v(5) === cons(v(8), v(9)))) <--> âˆƒ v(5) ; v(8) ; v(9) . v(5) and ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (v(5) === cons(v(8), v(9)))) [e-set] ;

(35)((âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (âˆƒ v(5) . v(5) and (v(5) === cons(v(8), v(9))))) <--> âˆƒ v(5) ; v(8) ; v(9) . v(5) and ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (v(5) === cons(v(8), v(9)))) [eqv-tranz(34, 33)] ;

(36)(âˆƒ v(5) . (cons(v(8), v(9)) <--> v(5) and (v(5) === cons(v(8), v(9))))) [e-subst] ;

(37)((cons(v(8), v(9)) <--> âˆƒ v(5) . v(5) and (v(5) === cons(v(8), v(9)))) <--> âˆƒ v(5) . (cons(v(8), v(9)) <--> v(5) and (v(5) === cons(v(8), v(9))))) [e-scope] ;

(38)(cons(v(8), v(9)) <--> âˆƒ v(5) . v(5) and (v(5) === cons(v(8), v(9)))) [modus-ponens(36, 37)] ;

(39)((âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and cons(v(8), v(9))) <--> âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (âˆƒ v(5) . v(5) and (v(5) === cons(v(8), v(9))))) [e-context(38)] ;

(40)((âˆƒ .Vars . âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and cons(v(8), v(9))) <--> âˆƒ .Vars . âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (âˆƒ v(5) . v(5) and (v(5) === cons(v(8), v(9))))) [e-context(39)] ;

(41)((âˆƒ .Vars . âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (âˆƒ v(5) . v(5) and (v(5) === cons(v(8), v(9))))) <--> âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (âˆƒ v(5) . v(5) and (v(5) === cons(v(8), v(9))))) [e-set] ;

(42)((âˆƒ .Vars . âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and cons(v(8), v(9))) <--> âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and cons(v(8), v(9))) [e-set] ;

(43)((âˆƒ .Vars . âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and cons(v(8), v(9))) <--> âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (âˆƒ v(5) . v(5) and (v(5) === cons(v(8), v(9))))) [eqv-tranz(41, 40)] ;

(44)((âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and cons(v(8), v(9))) <--> âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (âˆƒ v(5) . v(5) and (v(5) === cons(v(8), v(9))))) [eqv-tranz(43, 42)] ;

(45)((âˆƒ v(5) . v(5) and ((v(5) === cons(a, z)) or (v(5) === cons(x, a)))) <--> âˆƒ v(5) ; v(8) ; v(9) . v(5) and ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (v(5) === cons(v(8), v(9)))) [eqv-tranz(27, 31)] ;

(46)((âˆƒ v(5) . v(5) and ((v(5) === cons(a, z)) or (v(5) === cons(x, a)))) <--> âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and (âˆƒ v(5) . v(5) and (v(5) === cons(v(8), v(9))))) [eqv-tranz(45, 35)] ;

(47)((âˆƒ v(5) . v(5) and ((v(5) === cons(a, z)) or (v(5) === cons(x, a)))) <--> âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and cons(v(8), v(9))) [eqv-tranz(44, 46)] ;

(48)(cons(a, z) or cons(x, a) <--> âˆƒ v(8) ; v(9) . ((v(8) === a) and (v(9) === z) or (v(8) === x) and (v(9) === a)) and cons(v(8), v(9))) [eqv-tranz(12, 47)] ;
[94mChecked: [0m [1m[92m true
[0m
